storage/innobase/dict/dict0dict.cc

```diff
@@ -958,11 +958,12 @@ void dict_sys_t::lock_wait(SRW_LOCK_ARGS(const char *file, unsigned line))
   if (latch_ex_wait_start.compare_exchange_strong
       (old, now, std::memory_order_relaxed, std::memory_order_relaxed))
   {
+#ifdef UNIV_DEBUG
+    latch.x_lock(SRW_LOCK_ARGS(file, line));
+#else
     latch.wr_lock(SRW_LOCK_ARGS(file, line));
+#endif
```

Comment:
Sure, I will try to revise efb6b689bcd0ea9e0c5aa0c53efb432c78e34977 in #3021 to address this. Possibly, there could be a `srw_lock` compatible wrapper of `sux_lock` that would be used in debug builds. This is a little challenging to arrange, because `srw_lock` is a building block of `sux_lock`.

Reply:
For #3021 I implemented `srw_lock_debug` and made not only `dict_sys_t::latch` but also `trx_rseg_t::latch`, `lock_sys_t::latch` and `fil_space_t::latch` use it. This allows us to remove some debug code for tracking ownership. The tracking of shared lock holders was incomplete.